File: data_structure&algorithm/etc/recurvie_function/sum-list.js
let numbers = [3, 1, 4, 1, 5, 9];

function recursive(acc, array) {
    if (array.length === 0) {
        console.log(`total sum is ${acc}`)
        return acc
    } else {
        try {
            console.log(`recursive(${acc} [${array}]) is "`)
            return recursive(acc + array[0], array.slice(1));
        } catch (e) {} finally {
            console.log('"is what they said.')
        }
    }
}

recursive(0, numbers)

File: etc/recurvie_function/python/sum-list.py
numbers = [3, 1, 4, 1, 5, 9]


def resursive(acc, array):
    if len(array) == 0:
        print("Total sum is {}".format(acc))
        return acc
    else:
        return resursive(acc + array[0], array[1:])


resursive(0, numbers)
